vendors/nuvoton/boards/numaker_iot_m487_wifi/aws_demos/application_code/nuvoton_code/entropy_hardware_poll.c

```diff
@@ -189,9 +212,11 @@ int mbedtls_hardware_poll( void *data,
 #else
     unsigned char tmpBuff[PRNG_KEY_SIZE];
     size_t cur_length = 0;
+    *olen = 0;
     ((void) data);
 
-    init_adc_init();
+    if( trng_init() == pdFALSE ) return (-1);
```

Comment:
This call could potentially race if there are two or more application threads that create an mbedTLS RNG context (i.e. for the first time) at the same time. Could the RNG init routine be moved to main?

Reply:
@dcgaws , @qiutongs , Fixed it by critical section  and verified by multi-task test, in the future, maybe init & de_init in one transaction for power saving.

commit at: 2f2bb1b of the [fork-repo](https://github.com/cyliangtw/amazon-freertos-share/tree/feature/m487_cmake_0531)